lib/sauce-launch-service.js

```diff
@@ -21,9 +21,17 @@ class SauceLaunchService {
         const sauceConnectTunnelIdentifier = this.sauceConnectOpts.tunnelIdentifier
 
         if (sauceConnectTunnelIdentifier) {
-            capabilities.forEach((capability) => {
-                capability.tunnelIdentifier = capability.tunnelIdentifier || sauceConnectTunnelIdentifier
-            })
+            if (Array.isArray(capabilities)) {
+                capabilities.forEach((capability) => {
+                    capability.tunnelIdentifier = capability.tunnelIdentifier || sauceConnectTunnelIdentifier
+                })
+            } else {
+                Object.keys(capabilities).forEach((browser) => {
+                    if (capabilities[browser].desiredCapabilities) {
```

Comment:
Why should `capabilities[browser].desiredCapabilities` ever be undefined?

Reply:
I suppose it shouldn't, had added it just to be safe. Can remove if you prefer.

Reply:
> Can remove if you prefer.

Yes, if we don't expect this to be undefined let not check for it

Reply:
Sure, removed it.